pkg/daemons/control/deps/deps.go

```diff
@@ -94,6 +94,11 @@ func KubeConfig(dest, url, caCert, clientCert, clientKey string) error {
 	}
 	defer output.Close()
 
+	// cis-1.24 and newer require kubeconfigs to be 0600
+	if err := output.Chmod(0600); err != nil {
```

Comment:
We're using `os.Create` up above, which is a [one-line wrapper](https://cs.opensource.google/go/go/+/refs/tags/go1.20.6:src/os/file.go;l=314) around `os.OpenFile`: 
```golang
OpenFile(name, O_RDWR|O_CREATE|O_TRUNC, 0666)
````

It would probably be more efficient to just make that call directly ourselves, with the correct permissions (0600 instead of 0666)

Reply:
Makes sense